pkg/config/validation/validation.go

```diff
@@ -3477,3 +3479,57 @@ func validateTelemetryProviders(providers []*telemetry.ProviderRef) error {
 	}
 	return nil
 }
+
+// ValidateWasmPlugin validates a WasmPlugin.
+var ValidateWasmPlugin = registerValidateFunc("ValidateWasmPlugin",
+	func(cfg config.Config) (Warning, error) {
+		spec, ok := cfg.Spec.(*extensions.WasmPlugin)
+		if !ok {
+			return nil, fmt.Errorf("cannot cast to service entry")
+		}
+
+		errs := Validation{}
+		errs = appendValidation(errs,
+			validateWorkloadSelector(spec.Selector),
+			validateWasmPluginURL(spec.Url),
+			validateWasmPluginSHA(spec),
+		)
+		return errs.Unwrap()
+	})
+
+func validateWasmPluginURL(pluginURL string) error {
+	if pluginURL == "" {
+		return fmt.Errorf("url field needs to be set")
+	}
+	validSchemes := map[string]bool{
+		"": true, "file": true, "http": true, "https": true, "oci": true,
+	}
+
+	u, err := url.Parse(pluginURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse url: %s", err)
+	}
+	if _, found := validSchemes[u.Scheme]; !found {
+		return fmt.Errorf("url contains unsupported scheme: %s", u.Scheme)
+	}
+	return nil
+}
+
+func validateWasmPluginSHA(plugin *extensions.WasmPlugin) error {
+	if plugin.XSha256 == nil {
+		if strings.HasPrefix(plugin.Url, "http") {
+			return fmt.Errorf("sha256 field must be set when fetching .wasm module from HTTP(s) URL")
+		}
+		return nil
+	}
+	if len(plugin.GetSha256()) != 64 {
+		return fmt.Errorf("sha256 field must be 64 characters long")
+	}
+	validCharacters := "0123456789abcdef"
+	for _, char := range plugin.GetSha256() {
+		if !strings.ContainsRune(validCharacters, char) {
+			return fmt.Errorf("sha256 field must be alphanumeric: %s", plugin.GetSha256())
```

Comment:
alphanumeric sometimes includes capital letters (A-Z) so maybe we should be more explicit in this comment: like "must match [a-z0-9] pattern"

Reply:
error message has been adjusted